src/apply_conf.rs

```diff
@@ -183,29 +183,36 @@ fn copy_connection_files(
 
     for interface in &host.interfaces {
         info!("Processing interface '{}'...", &interface.logical_name);
+        let connections = &interface
+            .connection_ids
+            .clone()
+            .unwrap_or(vec![interface.logical_name.clone()]);
 
-        let mut filename = &interface.logical_name;
+        for connection in connections {
+            info!("Processing connection '{}'...", connection);
+            let mut filename = connection.clone();
 
-        let filepath = keyfile_path(host_config_dir, filename)
-            .ok_or_else(|| anyhow!("Determining source keyfile path"))?;
+            let filepath = keyfile_path(host_config_dir, &filename)
+                .ok_or_else(|| anyhow!("Determining source keyfile path"))?;
 
-        let mut contents = fs::read_to_string(filepath).context("Reading file")?;
+            let mut contents = fs::read_to_string(filepath).context("Reading file")?;
 
-        // Update the name and all references of the host NIC in the settings file if there is a difference from the static config.
-        match local_interfaces.get(&interface.logical_name) {
-            None => {}
-            Some(local_name) => {
-                info!(
-                    "Using interface name '{}' instead of the preconfigured '{}'",
-                    local_name, interface.logical_name
-                );
-
-                contents = contents.replace(&interface.logical_name, local_name);
-                filename = local_name;
+            // Update the name and all references of the host NIC in the settings file if there is a difference from the static config.
+            match local_interfaces.get(&interface.logical_name) {
+                None => {}
+                Some(local_name) => {
+                    info!(
+                        "Using interface name '{}' instead of the preconfigured '{}'",
+                        local_name, interface.logical_name
+                    );
+
+                    contents = contents.replace(&interface.logical_name, local_name);
+                    filename = filename.replace(&interface.logical_name, local_name);
```

Comment:
Can we cover this in the tests?

Reply:
Let me try...

Reply:
Covered in `copy_connection_files_successfully`